front/src/modules/trainschedule/components/ImportTimetableItem/generateTrainSchedulesPayloads.ts

```diff
@@ -20,50 +21,128 @@ export function generateTrainSchedulesPayloads(
 
     const departureTime = new Date(train.departureTime);
 
-    const { path, schedule } = train.steps.reduce(
+    const { path, schedule } = train.steps.reduce<{
+      path: TrainSchedule['path'];
+      schedule: NonNullable<TrainSchedule['schedule']>;
+    }>(
       (acc, step, index) => {
         const stepId = nextId();
-
         // Conditionally skip invalid UIC or CH code steps
         if (checkChAndUIC && !step.uic) {
           console.error(`Invalid UIC for step ${step.name}`);
-          return acc; // Skip invalid step
+          return acc;
         }
         if (checkChAndUIC && !step.chCode) {
           console.error(`Invalid CH code for step ${step.name}`);
-          return acc; // Skip invalid step
+          return acc;
+        }
+
+        if (!step.uic && step.trigram) {
+          acc.path.push({
+            id: stepId,
+            trigram: step.trigram,
+            secondary_code: step.chCode,
+          });
+        } else {
+          acc.path.push({
+            id: stepId,
+            uic: Number(step.uic),
+            secondary_code: step.chCode,
+          });
         }
-        if (!step.uic && step.trigram)
-          acc.path.push({ id: stepId, trigram: step.trigram, secondary_code: step.chCode });
-        else acc.path.push({ id: stepId, uic: Number(step.uic), secondary_code: step.chCode });
 
-        // Skip first step, handle time differences
         if (index !== 0) {
           const arrivalTime = new Date(step.arrivalTime);
-          const schedulePoint: NonNullable<TrainSchedule['schedule']>[number] = {
+          acc.schedule.push({
             at: stepId,
             arrival: Duration.subtractDate(arrivalTime, departureTime).toISOString(),
             stop_for: step.duration ? `PT${step.duration}S` : undefined,
-          };
-          acc.schedule.push(schedulePoint);
+          });
         }
 
         return acc;
       },
       {
-        path: [] as TrainSchedule['path'],
-        schedule: [] as NonNullable<TrainSchedule['schedule']>,
+        path: [],
+        schedule: [],
       }
     );
 
+    const basePayload = {
+      path,
+      train_name: train.trainNumber,
+      rolling_stock_name: train.rollingStock || '',
+      constraint_distribution: 'MARECO' as Distribution,
+      schedule,
+      start_time: departureTime.toISOString(),
+    };
+
+    payloads.push(basePayload);
+    return payloads;
+  }, []);
+}
+
+export function generatePacedTrainPayloads(
+  trains: ImportedPacedTrainSchedule[],
+  checkChAndUIC = true
+): PacedTrain[] {
+  return trains.reduce<PacedTrain[]>((payloads, train) => {
```

Comment:
We discussed out-of-band, and we figured we should be able to get rid of the code duplication here :)